presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1104,34 +1107,32 @@ else if (criteria instanceof JoinOn) {
 
             for (Expression conjunct : ExpressionUtils.extractConjuncts((Expression) optimizedExpression)) {
                 conjunct = ExpressionUtils.normalize(conjunct);
```

Comment:
I think `NOT(NOT(a.x = b.x))` is pretty unlikely. That said I *think* it will just work if you move it to `SimplifyExpressions`. What should happen is that `SimplifyExpressions` will change it to `a.x = b.x` and then the `PredicatePushDown` optimizer should move it into the `JoinNode` as an `EquiJoinClause`. Be sure to test though, as I'm less familiar with the push down optimizer.

In general, I think the planners should be designed to generate the simplest possible plan which is correct, so I'd even go so far as to say that the planner shouldn't be putting anything in the `EquiJoinClause` and should plan it as a `CROSS JOIN` followed by a `Filter`, and then leave it to the `PredicatePushDown` optimizer to "fix" it.

This has the advantage that even if I write my query as:
```
SELECT *
FROM a
CROSS JOIN b
WHERE a.x = b.x
```
The optimizer could convert it to an `INNER JOIN` (I don't think we do this yet)